### PR TITLE
Link/unlink device calls

### PR DESCRIPTION
**Pull request by rubdos**
Calls intended to run on primary devices to allow and delete secondary devices. #59 contd.

### CODE REVIEW

**Comment by gferon** [comment]
Is this ready to merge? 🚀

**Comment by rubdos** [comment]
> Is this ready to merge? rocket

Not tested, I'll merge when it's tested. Looking good so far?

**Comment by rubdos** [comment]
@gferon Few possibly weird design choices, could you specifically have a look at my change in Actix' `push_service.rs`? Would like a second opinion there :-)

File: libsignal-service/src/provisioning.rs [hunk 1]
```diff
@@ -310,3 +408,33 @@ impl<WS: WebSocketService> ProvisioningPipe<WS> {
         combined.filter_map(|x| async { x })
     }
 }
+
+#[cfg(test)]
```
Comment (gferon) [inline libsignal-service/src/provisioning.rs:412]: amazing 🥳

**Review by gferon — APPROVED** [review]
LGTM, awesome!

File: libsignal-service/src/provisioning.rs (hunk 1 above)
Comment (rubdos) [reply libsignal-service/src/provisioning.rs:412]: :tada:

File: libsignal-service-actix/src/push_service.rs [hunk 2]
```diff
@@ -83,6 +83,68 @@ impl PushService for AwcPushService {
...
+        log::debug!("AwcPushService::delete response: {:?}", response);
+
+        Self::from_response(&mut response).await?;
+
+        // In order to debug the output, we collect the whole response.
+        // The actix-web api is meant to used as a streaming deserializer,
+        // so we have this little awkward switch.
+        //
+        // This is also the reason we depend directly on serde_json, however
+        // actix already imports that anyway.
```
[the remark this answers is not in the document]
Comment (rubdos) [reply libsignal-service-actix/src/push_service.rs:122]: Fixed every comment but this.  Move to #63, then I can merge ;-)

**Pull request merged**